vis-lua.c

```diff
@@ -3219,20 +3220,23 @@ void vis_lua_process_response(Vis *vis, const char *name,
 	vis_lua_event_get(L, "process_response");
 	if (lua_isfunction(L, -1)) {
 		lua_pushstring(L, name);
-		switch (rtype) {
-		case EXIT:
-		case SIGNAL:
-			lua_pushinteger(L, len);
-			break;
-		default: lua_pushlstring(L, buffer, len);
-		}
 		switch (rtype){
 		case STDOUT: lua_pushstring(L, "STDOUT"); break;
 		case STDERR: lua_pushstring(L, "STDERR"); break;
 		case SIGNAL: lua_pushstring(L, "SIGNAL"); break;
 		case EXIT: lua_pushstring(L, "EXIT"); break;
 		}
-		pcall(vis, L, 3, 0);
+		switch (rtype) {
+		case EXIT:
+		case SIGNAL:
+			lua_pushinteger(L, len);
+			lua_pushlstring(L, buffer, 0);
+			break;
+		default:
+			lua_pushinteger(L, 0);
```

Comment:
In this case `lua_pushnil(L)` is more appropriate since the program
hasn't exited yet and `0` represents a normal exit in general.